servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultSingleAddressHttpClientBuilder.java

```diff
@@ -263,8 +255,9 @@ public StreamingHttpClient buildStreaming() {
         // Track resources that potentially need to be closed when an exception is thrown during buildStreaming
         final CompositeCloseable closeOnException = newCompositeCloseable();
         try {
-            final SdStatusCompletable sdStatus = new SdStatusCompletable();
-            final Publisher<? extends ServiceDiscovererEvent<R>> sdEvents = ctx.discover(sdStatus);
+
+            final Publisher<ServiceDiscovererEvent<R>> sdEvents =
+                    ctx.serviceDiscoverer.discover(ctx.address()).flatMapConcatIterable(identity());
```

Comment:
If we change public API of SD to emit `List` of addresses, should we also change LB factory API to use the same API for `eventPublisher`? Looks a bit weird do have these API different.

Reply:
I think sending the `List` to load balancer does not have any value as LBs are not expected to cache, batch SD updates. Passing the `List` makes the API more complex, so I decided not to propagate the changes to the LB layer. If we see a need in the future, it should be straight forward to change.